app/experimenter/static/js/components/DesignForm.js

```diff
@@ -28,11 +28,18 @@ class DesignForm extends React.PureComponent {
   }
 
   getEndpointUrl() {
-    return `experiments/${this.props.slug}/design-${this.props.experimentType}/`;
+    if (this.state.data.get("is_branched_addon")) {
+      return `experiments/${this.props.slug}/design-branched-addon/`;
+    }
+    return `experiments/${this.props.slug}/design-${this.state.data.get(
+      "type",
+    )}/`;
   }
 
   async componentDidMount() {
-    const data = await makeApiRequest(this.getEndpointUrl());
+    const data = await makeApiRequest(
+      `experiments/${this.props.slug}/design-${this.props.experimentType}/`,
+    );
```

Comment:
1) we can consolidate this if we don't treat branched-addon as a type, but rather pass the bool in separate as a prop and just check it in getEndpointURL
2) is_branched_addon can come from either the page prop or the state, and we should prefer the state